src/instructlab/model/download.py

```
@@ -1,31 +1,258 @@
+        super().__init__(
+            repository=repository, release=release, download_dest=download_dest
+        )
+        self.ctx = ctx
+
+    def _build_oci_model_file_map(self, oci_model_path: str) -> dict:
+        """
+        Helper function to build a mapping between blob files and what they represent
+        Format for the index.json file can be found here: https://github.com/opencontainers/image-spec/blob/main/image-layout.md#indexjson-file
+        """
+        index_hash = ""
+        index_ref_path = f"{oci_model_path}/index.json"
+        try:
+            index_ref = _load_json(Path(index_ref_path))
+            match = None
+            for manifest in index_ref["manifests"]:
+                if (
+                    manifest["mediaType"]
+                    == "application/vnd.oci.image.manifest.v1+json"
+                ):
+                    match = _extract_SHA(manifest["digest"])
+
+            if match:
+                index_hash = match.group(1)
+            else:
+                raise ValueError(
+                    f"could not find hash for index file at: {oci_model_path}"
+                )
+        except Exception as exc:
+            raise exc
+
+        blob_dir = f"{oci_model_path}/blobs/sha256"
+        try:
+            index = _load_json(Path(f"{blob_dir}/{index_hash}"))
+        except Exception as exc:
+            raise exc
+
+        title_ref = "org.opencontainers.image.title"
+        oci_model_file_map = {}
+        try:
+            for layer in index["layers"]:
+                match = _extract_SHA(layer["digest"])
+
+                if match:
+                    blob_name = match.group(1)
+                    oci_model_file_map[blob_name] = layer["annotations"][title_ref]
+        except Exception as exc:
+            raise ValueError(
+                f"failed to build OCI model file mapping from: {blob_dir}/{index_hash}"
+            ) from exc
+
+        return oci_model_file_map
```

(Sorry I haven't noticed it before). So if e.g. no layer matched in line 180, then the returned dictionary will be empty. And then `download` will do nothing, silently returning (instead of e.g. logging an error.) Should we raise an error here, or in `download`, when the dictionary is empty?